python/lsst/cp/pipe/ptc.py

```diff
@@ -588,7 +604,80 @@ def _makeZeroSafe(self, array, warn=True, substituteValue=1e-9):
         array[array == 0] = substituteValue
         return array
 
-    def fitPtcAndNl(self, dataset, ptcFitType):
+    def calculateLinearityResidualAndLinearizers(self, exposureTimeVector, meanSignalVector):
+        """Fit a quadratic polynomial to the mean vs time curve to get c0 for LinearizeSquared,
```

Comment:
1) I agree and was debating about this. In the end, I decided to fit the quadratic polynomial too because I thought that "LinearizeSquared" assumed that this is exactly the case (i.e, the c0 coefficient comes from exactly a quadratic fit, not a polynomial of another order). However, as you point out, I can just stick to the general n-order polynomial, still approximate c0 to -k2/(k1^2), and check that the k[3:] are ~zero (issuing a warning if for some reason they are not). 
2) Something for another ticket? 
3) As in 2), perhaps this could be done in another ticket. I think you are correct, in the sense that, for example, we would like to use information from a calibration diode to correct for signal-chain non-linearity. For this ticket, we agreed (in Jira) that, for the moment, we would do something simple like taking the linear part of an n-degree polynomial as the linearity correction, and that we would put that information in a lookup table.

Reply:
Ok, I think I wasn't sure how much had been delegated to the future/alternate linearity functions/etc.